wot-wg-2023-draft.html

```diff
@@ -206,98 +206,42 @@ <h2>Scope</h2>
-            Add support for describing historical values or value changes of affordances to the TD specification in an
-            interoperable way with existing solutions.
-          </dd>
-          <dt>
-            <strong>Connection and Payload Reuse Improvements</strong>
-            (<a href="#thing-description-deliverable">Thing Description</a>,
-            <a href="#binding-deliverable">Binding Templates</a>,
-            <a href="#profiles-deliverable">Profiles</a>):
-          </dt>
-          <dd>
-            Add support for describing cases where a connection and subsequent payloads are same or similar.
-          </dd>
-          <dt>
-            <strong>Manageable Actions</strong>
-            (<a href="#thing-description-deliverable">Thing Description</a>,
-            <a href="#profiles-deliverable">Profiles</a>):
-          </dt>
-          <dd>
-            Add support for describing actions that can be managed after their invocation.
+            <a href="#profiles-deliverable">Profiles</a>,
+            <a href="#binding-deliverable">Binding Templates</a>):</dt>
+          <dd>The WG will improve the usability of WoT in other ecosystems and communities. Thus, the WG will define core binding and profiling mechanisms and specify additional protocol and payload bindings, and profiles together with reusable, external security vocabularies.
           </dd>
           <dt>
-            <strong>TD Versioning</strong>
+            <strong>Improve Management of TDs</strong>
             (<a href="#thing-description-deliverable">Thing Description</a>,
             <a href="#discovery-deliverable">Discovery</a>):
           </dt>
           <dd>
-            Explain the semantics on how to version Thing Descriptions.
+            The Thing Descriptions require additional mechanisms around them for their management.
+            The working group will update the Discovery specification with additional features, add mechanism to sign TDs, how to manage versions of TDs and how to parse, validate, consume and lint TDs.
           </dd>
           <dt>
-            <strong>Normative Linting, Parsing, Consuming and Validation mechanism for TDs</strong>
+            <strong>Increase descriptiveness of TDs</strong>
             (<a href="#thing-description-deliverable">Thing Description</a>,
-            <a href="#scripting-deliverable">Scripting API</a>,
-            <a href="#profiles-deliverable">Profiles</a>):
+            <a href="#binding-deliverable">Binding Templates</a>):
           </dt>
           <dd>
-            Define a linting mechanism for Thing Descriptions that allows adding additional rules and normative mechanism for
-            parsing, consuming and validation of TDs for Consumers.
+            The WG will work on increasing the descriptive power of TDs to allow using them in more use cases. This includes supporting historical values (timeseries), repeating payloads and connections, and manageable actions.
```

Comment:
Which use cases are you referring to? 
Not sure what is meant with "repeating payloads"?

Reply:
This is basically when there are repeated payloads and this results in repeated data schemas in TDs. You can see this in ecosystems/standards like ECHONET Lite Web API, Philips Hue or the HTTP Baseline Profile which requires repeated payloads for actions. 

In the case of HTTP Baseline Profile TDs, the `additionalResponses` is used which is sort semantically wrong. Additionally, even if we have `schemaDefinitions` in the Thing-level, we do not have a way to point to the "relevant" part of the payload for that affordance.

Reply:
I had assumed that "repeating payloads and connections" was in reference to protocols like WebSockets which currently can't be effectively described in a Thing Description. For that I actually prefer "Connection and Payload Reuse" from the current text, although in that case the connection re-use is the important thing. I think the payloads for WebSockets have to be defined in a `subprotocol`.

@egekorkan wrote:
> In the case of HTTP Baseline Profile TDs, the additionalResponses is used which is sort semantically wrong. Additionally, even if we have schemaDefinitions in the Thing-level, we do not have a way to point to the "relevant" part of the payload for that affordance.

The issue with action queues is not really "repeated" payloads and we already have operation names for describing "manageable actions". The missing pieces are:
1. Being able to describe dynamically created resources
2. Being able to define a data schema for a response to an invokeaction request which is separate from the action output schema

The latter is an example of a much broader problem which is that the TD spec needs to be much clearer about how to define data schemas for each type of operation, including meta operations (see https://github.com/w3c/wot-thing-description/issues/1665 for a detailed breakdown).

If this all needs to be summarised in a single sentence I would personally word it something like:

"This includes describing dynamic resources, re-usable connections, historical values and more precise definitions of data schemas for operations."

---

Side note: Thing Descriptions using the HTTP Basic Profile do not use `additionalResponses` or `schemaDefinitions` to define action payloads, the `ActionStatus` payload format is defined in the protocol binding specification in the profile and is assumed by conformant Consumers.

Reply:
@benfrancis I will change it based on your comments. I agree with them.

Side note on the side note: https://github.com/w3c/wot-testing/blob/main/data/input_2022/Profile/TD/Oracle/TDs/BluePump%20WebThing.td.jsonld is using those terms

Reply:
> Side note on the side note: https://github.com/w3c/wot-testing/blob/main/data/input_2022/Profile/TD/Oracle/TDs/BluePump%20WebThing.td.jsonld is using those terms

🤦 That extra metadata is completely redundant. I guess there's no reason you can't provide it, but it's only useful for Consumers which don't implement the profile. I would expect Consumers conforming to the profile to ignore it.

The only reason we were trying to describe the protocol binding declaratively was to illustrate it as an informative Thing Model in the WoT Profile specification (like the WoT Discovery specification does), which turned out not to be possible.

Reply:
Note: It may also be necessary to add new operation types to the next version of WoT Thing Description, like `subscribemultipleproperties`, `observemultipleproperties`, `observeaction`, `observeallactions` etc., but that might be covered under the general topic of "increasing the descriptive power of TDs". I'm assuming the list is not exhaustive.